data_steward/validation/ehr_merge.py

```diff
@@ -0,0 +1,281 @@
+    if len(incomplete_jobs) == 0:
+        query_result = bq_utils.get_job_details(person_mapping_query_job_id)
+        if 'errors' in query_result['status']:
+            logging.error('{} load failed!'.format(PERSON_ID_MAPPING_TABLE))
+            return "visit mapping table failed"
+    else:
+        logging.error('{} load taking too long!'.format(PERSON_ID_MAPPING_TABLE))
+        return "visit mapping table taking too long"
+
+    # # list of hpos with visit table and creating visit id mapping table queries
+    visit_hpo_queries = []
+    for hpo in hpos_with_visit:
+        visit_hpo_queries.append(VISIT_ID_HPO_BLOCK % locals())
+    union_all_blocks = '\n UNION ALL'.join(visit_hpo_queries)
+    visit_mapping_query = VIST_ID_MAPPING_QUERY_SKELETON % locals()
+
+    logging.info('Loading ' + VISIT_ID_MAPPING_TABLE)
+    query_result = query(visit_mapping_query,
+                         destination_table_id=VISIT_ID_MAPPING_TABLE,
+                         write_disposition='WRITE_TRUNCATE')
+    visit_mapping_query_job_id = query_result['jobReference']['jobId']
+    incomplete_jobs = bq_utils.wait_on_jobs([], retry_count=10)
+    if len(incomplete_jobs) == 0:
+        query_result = bq_utils.get_job_details(visit_mapping_query_job_id)
+        if 'errors' in query_result['status']:
+            logging.error('{} load failed!'.format(VISIT_ID_MAPPING_TABLE))
+            return "visit mapping table failed"
+    else:
+        logging.error('{} load taking too long!'.format(VISIT_ID_MAPPING_TABLE))
+        return "visit mapping table failed"
+
+    jobs_to_wait_on = []
+    for table_name in common.CDM_TABLES:
+        q = construct_query(table_name, hpos_to_merge, hpos_with_visit, project_id, dataset_id)
+        logging.info('Merging table: ' + table_name)
+        query_result = query(q, destination_table_id='merged_'+table_name, write_disposition='WRITE_TRUNCATE')
+        query_job_id = query_result['jobReference']['jobId']
+        jobs_to_wait_on.append(query_job_id)
+
+    incomplete_jobs = bq_utils.wait_on_jobs(jobs_to_wait_on, retry_count=10)
+    required_tables_flag = True
+    if len(incomplete_jobs) == 0:
+        table_errors = []
+        for job_id in jobs_to_wait_on:
+            query_result = bq_utils.get_job_details(job_id)
+            if 'errors' in query_result['status']:
+                table_errors.append(table_name)
+                if table_name in common.REQUIRED_TABLES:
+                    required_tables_flag = False
+        if len(table_errors) == 0:
+            logging.info(" ---- Merge succesful! ---- ")
+            return "success: " + ','.join(hpos_to_merge)
+        else:
+            logging.info(" ---- Following tables fail --- " + ",".join(table_errors))
+            if not required_tables_flag:
+                return "required-not-done"
+            return "required-done"
+    else:
+        raise RuntimeError("---- Merge takes too long! ---- ")
```

Comment:
I like that you are raising an error here, it seems like the right thing to do. If in addition we raise an error that wraps the failed job details, it presents an opportunity to give key information to those troubleshooting this. For this reason I'd argue that we should also raise an error if any of `jobs_to_wait_on` has status error [above](https://github.com/all-of-us/curation/pull/43/files#diff-e369fecc3f82add579c22aa3a4619724R254). This may very well save our future selves the headache of troubleshooting a system that has reached what seems to be an incredibly ambiguous state.

Reply:
done.